src/main/java/max/command/CommandParser.java

```diff
@@ -0,0 +1,78 @@
+package max.command;
+
+import java.util.HashMap;
+
+public class CommandParser {
+
+    public CommandParser() {
+
+    }
+
+    public String[] splitIntoCommands(String rawCommand) {
+        // " -" splits the string into substrings
+        // if the substring starts with two dashes '--' and is preceded by whitespace
+        // For example: deadline my_task --by -xyz sunday-monday
+        // Will output: ["deadline my_task", "by -xyz sunday-monday"]
+        // For now, this does not account for accidental or adversarial '--' input at the start of words
+        // but this is an unlikely edge case that can be dealt with later
+        return rawCommand.split(" --");
+    }
+
+    public HashMap<String, String> getCommandPayloadMap(String[] commandList) {
+        // Each command can be processed into a subcommand and its corresponding text value
+        // For example: "deadline work on CS2113 --by Sunday"
+        // Should map to: <"deadline", "work on CS2113">, <by, Sunday>
+        HashMap<String, String> commandMap = new HashMap<>();
+        for (String cmdStr : commandList) {
+            String cmd = extractCommandFromSubcommand(cmdStr);
+            String text = extractTextFromSubcommand(cmdStr);
+            commandMap.put(cmd, text);
+        }
+        return commandMap;
+    }
+
+    public static String extractCommandFromSubcommand(String subcommand) {
+        // Subcommands have format <subcommand> <text_to_extract>
+        // Return the <subcommand>
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return "by"
+        String[] words = subcommand.split(" ");
+        return words[0];
```

Comment:
Try to avoid magic literals. There has been a few more instances in the rest of your code. You can use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.